src/plugins/data/common/search/expressions/date_range.ts

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { ExpressionFunctionDefinition, ExpressionValueBoxed } from 'src/plugins/expressions/common';
+
+export interface DateRange {
```

Comment:
a lot of this interfaces already exist, please use the existing ones where possible

Reply:
I double-checked all the interfaces, and I couldn't find ones that I could reuse in the newly added functions. Although, there is `CidrMaskIpRangeAggKey` and `RangeIpRangeAggKey` could be partially replaced by extending `Cidr` and `IpRange` accordingly. But I wouldn't do that because they are used in the aggregation functions, and that seems like the interfaces are partially compatible but actually don't have any logical relation.